engine.py

```diff
@@ -1,46 +1,60 @@
-import random
 import time
 
 from teams import Team
 from characters import *
 
 
 class Engine:
-    def __init__(self):
+    def __init__(self, team1_size: int, team2_size: int) -> None:
         self.teams_list = []
-        # self.team_order = random.randint(0, 1)  # randoms starting team
         self.rounds = 0
         self.program_execution_time = 0
+        self.set_teams(team1_size, team2_size)
 
-    def create_new_team(self, name):
+    def create_new_team(self, name: str) -> Team:
         """
         Creates Team and sets it's name
         :param name: str
         :return: str, teams name
         """
-        tmp_name = Team(name)
-        self.teams_list.append(tmp_name)
-        return tmp_name
+        tmp_team = Team(name)
+        self.teams_list.append(tmp_team)
+        return tmp_team
 
-    def run_game(self):
+    def set_teams(self, team1_size: int, team2_size: int) -> None:
+        # # creates two opposite team objects and gives them names
+        team1 = self.create_new_team('New York Gangs')
+        team2 = self.create_new_team('Pirates of the Caribbean')
+
+        # # creates characters for both teams and adds them to the teams
+        team1.team_generator(team1_size)
+        team2.team_generator(team2_size)
+
+        # # now one team is known by the other one
+        team1.opponent_team = team2
+        team2.opponent_team = team1
+        # print(f'team1:{len(team1.team)}, team1 opponents:{len(team1.opponent_team.team)}')
+
+    def run_game(self) -> None:
         start_time = time.time()
+        random.shuffle(self.teams_list)  # randoms starting team
         self.fight()
-        self.program_execution_time = time.time() - start_time
-        print(self.battle_summary())
+        self.program_execution_time = round(time.time() - start_time, 4)
+        # print(self.battle_summary())
 
-    def fight(self):
+    def fight(self) -> None:
         """
-        Takes character that is randomed to start fight, and attacks randomed character from opponents team
+        Takes next character from drawn team to start fight against drawn character from opponents team
         If Character has skills working on your own team it also takes round.
         After attack/support it changes the team to another one.
         :return: Stops attack if chosen character is stunned
         """
 
-        while all(game.teams_list):
-            tmp_list = self.teams_list.copy()
+        while all(self.teams_list):  # works until one team is dead
+            tmp_list = self.teams_list.copy()  # copy of teams list, for generator
 
             for team in tmp_list:
-                team.generator_field = team.find_attacking_character()
+                team.generator_field = team.find_attacking_character()  # sets generator 'object'
```

Comment:
Nie nazywaj pola cośtam_field
W tym miejscu można zrobić piękne property, które wyciągnie następnego bohatera <3

Reply:
Nie pilnujesz żeby metody robiły to co sugeruje ich nazwa.
Find... sugeruje, źe coś znajdzie i pewnie zwróci, a nie że stworzy generator.

Reply:
W tym miejscu można zrobić piękne property, które wyciągnie następnego bohatera <3
nie bardzo wiem gdzie

Reply:
chodzi CI o to, zeby w petli while zastapic team.generator
getterem?

Reply:
Nie - żeby tam gdzie wywołujemy bohatera funkcją next(generator) wykorzystać property. Nazwiemy sobie getter next_character i on nam bedzie wykonywał tego nexta i go zwracał